scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -526,10 +550,42 @@
      :backfill-jobs (mapv :job/uuid jobs-to-backfill)
      :matched-head? (matched? head-of-considerable)}))
 
+(defn task->mesos-message
+  [t]
+  (let [cmd (:command t)
+        resources (:resources t)]
+    {:name (:name t)
+     :task-id [{:value (:task-id t)}]
+     :slave-id [{:value (:slave-id t)}]
+     :resources (vec (remove nil?
+                             [{:type :value-scalar,
+                               :name "mem",
+                               :scalar (:mem resources)
+                               :role (:role t)}
```

Comment:
Thanks for pointing this out @wyegelwel .  I misunderstood how the "wildcard" resources are treated in allocations.  It seems that if Mesos offers your framework 8 wildcard (*) CPUS and 8 CPU's that can only be used by your framework's role, you can't simply accept 16 CPU's with your role... Mesos won't simply assign wildcard CPU's to any role. You need to tell Mesos specifically to allocate 8 wildcard CPU's and 8 role-specific CPU's.    

So, you're right, we need to change the Cook's own logic that decides which resources to accept - it's not simply a case of swapping out clj-mesos for Mesomatic.  

It seems to me that the "polite approach" in terms of impact on other frameworks would be to use up all role-specific resources before dipping into the wildcard resources.